quality_check/08-gene_lengths.Rmd

```diff
@@ -0,0 +1,293 @@
+compendium_technology_labels <- rep("MICROARRAY", length(sample_names))
+compendium_technology_labels[grep("lengthScaledTPM", sample_names)] <- "RNASEQ"
+
+# get the technology labels as a data.frame for plotting
+technology_df <- as.data.frame(compendium_technology_labels) 
+```
+
+```{r}
+# what genes are short? using 25th percentile
+short_genes <- zeroes_df %>%
+  dplyr::filter(gene_length <= quantile(zeroes_df$gene_length, 0.25)) %>%
+  dplyr::pull(Gene)
+```
+
+#### PCA - all genes
+
+First we'll look at PC1 and PC2 for all genes
+
+```{r}
+pca_results <- prcomp(t(exprs_mat))
+# plotting with autoplot for PCA
+autoplot(pca_results, data = technology_df, alpha = 0.5, 
+         colour = 'compendium_technology_labels', scale = 0) +
+  scale_color_manual(labels = c("MICROARRAY", "RNA-SEQ"), 
+                     values = c("#E69F00", "#56B4E9")) +
+  theme_bw() +
+  ggtitle("Test Compendium by Technology: All genes") +
+  theme(plot.title = element_text(hjust = 0.5, face = "bold")) +
+  guides(colour = guide_legend(title = "technology"))
+```
+
+```{r}
+ggsave(filename = file.path(plot_dir, 
+                            "small_test_compendium_PCA.png"),
+       plot = last_plot())
+```
+
+Pairs plot (PC1-5)
+
+```{r}
+png(file.path(plot_dir, "small_test_compendium_pairs_plot.png"))
+pairs(pca_results$x[, 1:5], col = dplyr::recode(compendium_technology_labels,
+                                                `MICROARRAY` = "#E69F00",
+                                                `RNASEQ` = "#56B4E9"),
+      main = "Test Compendium by Technology: All genes")
+dev.off()
+```
+
+#### Drop the "short" genes from the expression matrix
+
+```{r}
+length_filtered_exprs_mat <- exprs_mat[!rownames(exprs_mat) %in% short_genes, ]
+dim(length_filtered_exprs_mat)
+```
+
+Redo PCA with the short genes removed
+
+```{r}
+# PCA on the length filtered exprs mat
+filtered_pca_results <- prcomp(t(length_filtered_exprs_mat))
+
+# plotting with autoplot for PCA
+autoplot(filtered_pca_results, data = technology_df, alpha = 0.5, 
+         colour = 'compendium_technology_labels', scale = 0) +
+  scale_color_manual(labels = c("MICROARRAY", "RNA-SEQ"), 
+                     values = c("#E69F00", "#56B4E9")) +
+  theme_bw() +
+  ggtitle("Test Compendium by Technology: Short genes removed") +
+  theme(plot.title = element_text(hjust = 0.5, face = "bold")) +
+  guides(colour = guide_legend(title = "technology"))
+```
+
+```{r}
+ggsave(filename = file.path(plot_dir, 
+                            "gene_lengths_test_compendium_PCA.png"),
+       plot = last_plot())
+```
+
+Pairs plot (PC1-5)
+
+```{r}
+png(file.path(plot_dir, "gene_lengths_test_compendium_pairs_plot.png"))
+pairs(filtered_pca_results$x[, 1:5], 
+      col = dplyr::recode(compendium_technology_labels,
+                          `MICROARRAY` = "#E69F00",
+                          `RNASEQ` = "#56B4E9"),
+      main = "Test Compendium by Technology: Short genes removed")
+dev.off()
```

Comment:
You could consider printing out all your plots here in addition to saving them as png's. I just like it because it puts all the plots in one spot and I don't have to go look for them elsewhere.

Reply:
I'm going to keep as is like I did in an early PR because the dimensions of the plot in the notebook render it not particularly useful.